SO-4896: Numeric relationship values update

apeteri opened the pull request.

apeteri committed:
SO-4896: Merge integerValue and decimalValue into a single BigDecimal...

...field in RelationshipValue

apeteri committed:
SO-4896: Update OWL axiom conversion and reasoner taxonomy building

apeteri committed:
SO-4896: Update classification, change recording and saving changes

apeteri committed:
SO-4896: Update SNOMED CT tests related to concrete values

apeteri committed:
SO-4896: Add value type to equals and hashCode computations

apeteri committed:
SO-4896: Add support for MRCM range constraint filter (exact matches)

apeteri committed:
SO-4896: Add relationship value type changes to the revision hash

apeteri committed:
SO-4896: Fix relationship value types using MRCM constraints after...

...each effective time slice is imported

apeteri committed:
SO-4896: Restore original heuristic when parsing RF2 number literals

apeteri committed:
SO-4896: Enable JSON serialization of RelationshipValue instances

apeteri committed:
SO-4896: Add null check to resolve an LGTM violation

cmark commented on snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/entry/SnomedOWLRelationshipDocument.java:
```diff
@@ -58,8 +59,7 @@ public static SnomedOWLRelationshipDocument createFrom(final SnomedRelationship
 	private final String destinationId;
-	private final Integer integerValue;
-	private final Double decimalValue;
+	private final BigDecimal numericValue;
```
What about backward compatibility? This will break any existing deployment where the two removed fields were present. Could you please add them back as read-only fields? Thanks!

cmark commented on snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/index/entry/SnomedRelationshipIndexEntry.java:
```diff
@@ -430,8 +417,7 @@ public SnomedRelationshipIndexEntry build() {
 	private final String destinationId;
-	private final Integer integerValue;
-	private final Double decimalValue;
+	private final BigDecimal numericValue;
```
Same here. Please apply the usual backward-compatibility pattern in this document type as well.

apeteri committed:
SO-4896: Restore compatibility with documents indexed previously

Conversion happens at the next re-indexing. Query expressions cover both
forms.

cmark reviewed (approved):
LGTM!
